test/files.js

```diff
@@ -9,7 +9,10 @@ import all from 'it-all'
 import last from 'it-last'
 import { expect } from 'aegir/utils/chai.js'
 
-const SHARD_THRESHOLD = 1000
+const SHARD_THRESHOLD = 2000
+// Use this to help push the shard threshold over the go-ipfs defaults
+// without just adding lots of files
+const shardFilePadding = "".padStart(100, '0')
```

Comment:
@achingbrain we could figure out adding a configurable sharding threshold in go-ipfs. However, this seemed like it wasn't super painful and wouldn't make tests take too much longer. WDYT?

Reply:
Having a configurable sharding threshold makes things like the test added as part of https://github.com/ipfs/js-ipfs/pull/3936 more feasible.  One or two tests like this are not super painful, but twenty or thirty would be.

Reply:
It also means you can do the whole test in code instead of having to deal with importing car files to get HAMTs that are \*just so\*.  Doing it all in code means it's easy to run in environments without filesystems like browsers.